HHS/Head-Start-TTADP
# [TTAHUB-1034] Use lastest AR to set Objective status

## Description of change

Objective status should only be set by an approved AR that has the most recent 'endDate'.

## How to test

**Note:** Until [TTAHUB-1040](https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1040) is complete you might need to modify the db data or simply run the tests.

1. Create a report with goals and objectives (end date in the past)
2. Approve the report 
3. Ensure the objective statuses have been updated in the DB
4. Create another report using the same goals and objectives (with a more recent end date)
5. Approve the report
6. Ensure objective statues have been updated with the new objective statues
7. Reset the second report back to DRAFT
8. Ensure the objectives have been rolled back to the statuses of the first approved report
9. Reset the first report back to DRAFT
10. Ensure the objective statues have been rolled back to 'Not Started'.

 

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1034


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
